crates/rattler_conda_types/src/match_spec/parse.rs

```diff
@@ -380,22 +405,25 @@ fn matchspec_parser(
     let (input, _comment) = strip_comment(input);
     let (input, _if_clause) = strip_if(input);
 
-    // 2. Is the spec a tarball?
+    // 2. Is the spec a package file, then we can parse it as a url and use as is
     if is_package_file(input) {
-        let _url = match Url::parse(input) {
-            Ok(url) => url,
-            #[cfg(target_arch = "wasm32")]
-            Err(_) => return Err(ParseMatchSpecError::InvalidPackagePathOrUrl),
-            #[cfg(not(target_arch = "wasm32"))]
-            Err(_) => match PathBuf::from_str(input) {
-                Ok(path) => Url::from_file_path(path)
-                    .map_err(|_err| ParseMatchSpecError::InvalidPackagePathOrUrl)?,
-                Err(_) => return Err(ParseMatchSpecError::InvalidPackagePathOrUrl),
-            },
-        };
+        let url = PathBuf::from_str(input)
+            .map_err(|_| ParseMatchSpecError::InvalidPackagePathOrUrl)
+            .and_then(|_path| {
+                file_url::file_path_to_url(input)
+                    .map_err(|_| ParseMatchSpecError::InvalidPackagePathOrUrl)
+            })
+            .or_else(|_| parse_url(input))?;
+
+        let identifier = ArchiveIdentifier::try_from_url(&url)
+            .ok_or(ParseMatchSpecError::InvalidPackagePathOrUrl)?;
+        let file_name = identifier.to_file_name();
 
-        // TODO: Implementing package file specs
-        unimplemented!()
+        return Ok(MatchSpec {
+            url: Some(url),
+            file_name: Some(file_name),
```

Comment:
I would completely remove everything to do with the file_name. The user only specified a URL, we don't need to add more restrictions into the matchspec itself because it no longer matches the user's intent.